projectHashProposal.md

```diff
@@ -0,0 +1,18 @@
+# LSIF: ProjectWithHash
+
+Formulating and processing LSIF for large repositories can be very time consuming. We've discovered that many commits touch only ~20% of projects in a given repository, so there can be lots of unnecessary processing done on unchanged projects that are then ignored at ingestion time. To decrease processing times, I propose adding a new "projectHash" property to the existing [project vertex](https://github.com/Microsoft/language-server-protocol/blob/master/indexFormat/specification.md#the-project-vertex). The new definition for the `project` vertex would be:
+
+```ts
+export interface ProjectWithHash extends Project {
+    /**
+     * An optional hash of the files included in the project, `base64` encoded.
+     */
+    projectHash?: string;
+}
+```
+
+Individual languages can decide what is the most reasonable selection for this project hash. For C#, this could be the MVID of the project assembly. For typescript, this could be a hash of the contents of all files included in the tsconfig.json. This `projectHash` property can then be compared against an existing cache of project hashes for the given repository to determine if the content has changed and the project needs to be re-ingested.
+
+# Rich Code Navigation: Application
```

Comment:
We can generalize this statement for indexers in general and not make it specific to Rich Navigation.

Reply:
Great suggestion! Done